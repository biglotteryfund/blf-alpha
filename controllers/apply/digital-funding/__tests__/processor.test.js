/* eslint-env jest */
'use strict';
const fs = require('fs');
const nodemailer = require('nodemailer');
const path = require('path');
const yaml = require('js-yaml');
const { get } = require('lodash');

// Mock email secret
process.env.DIGITAL_FUNDING_EMAIL = 'digital@example.com';

const processor = require('../processor');
const formModel = require('../form-model');
const { flattenFormData, stepsWithValues } = require('../../helpers');

const langData = fs.readFileSync(path.resolve(__dirname, '../../../../config/locales/en.yml'), 'utf-8');
const enCopy = yaml.safeLoad(langData);

const mockFormData = {
    'step-1': {
        name: 'Example Person',
        email: 'example@example.com',
        'organisation-name': 'Test Organisation',
        'about-your-organisation': 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Porro, inventore.'
    }
};

function cleanMailForSnaphot(info) {
    info.message = info.message
        .toString()
        .replace(/Message-ID: .*\n/, '')
        .replace(/Date: .*\n/, '')
        .replace(/ boundary=.*\n/, '')
        .replace(/----.*\n/gm, '');
    return info;
}

describe('processor', () => {
    it('should create success emails', async () => {
        const mockTransport = nodemailer.createTransport({
            streamTransport: true,
            newline: 'unix',
            buffer: true
        });

        const form = formModel(1);
        const results = await processor({
            form: form,
            data: flattenFormData(mockFormData),
            stepsWithValues: stepsWithValues(form.steps, mockFormData),
            mailTransport: mockTransport,
            copy: get(enCopy, form.lang)
        });

        const [customerEmail] = results.map(cleanMailForSnaphot);
        expect(customerEmail.message).toMatchSnapshot();
    });
});
