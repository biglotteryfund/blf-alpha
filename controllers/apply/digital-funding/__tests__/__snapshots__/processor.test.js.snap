// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`processor should create success emails 1`] = `
"Content-Type: multipart/alternative;
From: Big Lottery Fund <noreply@biglotteryfund.org.uk>
To: Example Person <example@example.com>
Subject: Thank you for getting in touch with the Big Lottery Fund!
MIME-Version: 1.0

Content-Type: text/plain
Content-Transfer-Encoding: quoted-printable

Thank you for getting in touch with us.

Here is a copy of the information =
you've submitted. We'll be in touch as soon as we can.

YOUR INFORMATION
YOUR DETAILS
NAME
Example Person

EMAIL ADDRESS
example@example.com

ORGANISATION NAME
Test Organisation

ABOUT YOUR ORGANISATION
Lorem ipsum dolor sit amet consectetur adipisicing elit. Porro, inventore.

DATA PROTECTION
The Big Lottery Fund is a public body with a duty to =
distribute National Lottery and other money in grants for good causes. We =
use
the personal data you provide, such as contact details for individuals =
at your organisation, to help you apply for a grant and to
assess your application. We may share your personal data with organisations=
 which help us with our grant making activities or
others which have a =
legitimate interest in our work or have funded your grant. We will only =
share personal data which they need to
carry out their work and subject to =
appropriate safety measures. Our Data Protection and Privacy Notice gives =
more information
about how we store and use personal data and the lawful =
basis for this. Please read the full Notice which is published on our
website at www.biglotteryfund.org.uk/data-protection or contact us to =
request a hard copy. The Notice may be updated from time to
time.
Content-Type: text/html
Content-Transfer-Encoding: quoted-printable

<!doctype html>
<html>
    <head>
        <meta charset=3D\\"UTF-8\\">
       =20
        <style>
@font-face {
  font-family: 'Poppins';
  font-style: normal;
  font-weight: 300;
  src: local('Poppins Light'), =
local('Poppins-Light'), url(https://fonts.gstatic.com/s/poppins/v5/pxiByp8k=
v8JHgFVrLDz8Z1xlEA.ttf) format('truetype');}
@font-face {
  font-family: 'Poppins';
  font-style: normal;
  font-weight: 500;
  src: local('Poppins Medium'), local('Poppins-Medium'), url(https://fonts.=
gstatic.com/s/poppins/v5/pxiByp8kv8JHgFVrLGT9Z1xlEA.ttf) =
format('truetype');}
</style>
    </head>
    <body style=3D\\"margin: auto; =
max-width: 75%; padding: 20px; color: #333; font-family: arial; font-size: =
20px; font-weight: 400; line-height: 1.35;\\">
        <div class=3D\\"logo\\" =
style=3D\\"display: block; margin-bottom: 20px; text-align: right;\\">
            <img src=3D\\"https://www.biglotteryfund.org.=
uk/assets/images/logos/logo.png\\" alt=3D\\"Big Lottery Fund logo\\" =
style=3D\\"width: 150px;\\" width=3D\\"150\\">
        </div>

        <p style=3D\\"color: #333; font-family: arial; font-size: 20px; =
font-weight: 400; line-height: 1.35;\\">Thank you for getting in touch with =
us.</p>

        <p style=3D\\"color: #333; font-family: arial; font-size: =
20px; font-weight: 400; line-height: 1.35;\\">Here is a copy of the =
information you've submitted. We'll be in touch as soon as we can.</p>

        <h2 class=3D\\"space-above\\" style=3D\\"color: #e20087; font-weight: =
500; margin: 0 0 20px; font-size: 22px; margin-top: 50px;\\">Your =
information</h2>

        <div class=3D\\"summary\\">
           =20
                <h4 style=3D\\"color: #e20087; margin: 0 0 20px; font-size: =
16px; border-bottom: 1px solid #e20087; padding-bottom: 3px; font-weight: =
700;\\">Your details</h4>
               =20
                       =20
                       =20
                       =20
                            <h5 style=3D\\"color: #e20087; margin: 0 0 20px; =
font-size: 14px; font-weight: 700;\\">Name</h5>
                           =
=20
                                <p style=3D\\"color: #333; font-family: =
arial; font-size: 20px; font-weight: 400; line-height: 1.35;\\">Example =
Person</p>
                           =20
                       =20
                       =20
                       =20
                       =20
                            <h5 style=3D\\"color: =
#e20087; margin: 0 0 20px; font-size: 14px; font-weight: 700;\\">Email =
address</h5>
                           =20
                               =
 <p style=3D\\"color: #333; font-family: arial; font-size: 20px; font-weight:=
 400; line-height: 1.35;\\">example@example.com</p>
                         =
  =20
                       =20
                       =20
                       =20
                       =20
                            <h5 style=3D\\"color: #e20087; margin: 0 0 20px; =
font-size: 14px; font-weight: 700;\\">Organisation name</h5>
                           =20
                                <p =
style=3D\\"color: #333; font-family: arial; font-size: 20px; font-weight: =
400; line-height: 1.35;\\">Test Organisation</p>
                           =
=20
                       =20
                       =20
                       =20
                       =20
                            <h5 style=3D\\"color: #e20087; margin: 0 0 20px; =
font-size: 14px; font-weight: 700;\\">About your organisation</h5>
                           =20
                                <p =
style=3D\\"color: #333; font-family: arial; font-size: 20px; font-weight: =
400; line-height: 1.35;\\">Lorem ipsum dolor sit amet consectetur adipisicing=
 elit. Porro, inventore.</p>
                           =20
                       =20
           =20
        </div>

        <h5 class=3D\\"space-above\\" style=3D\\"color: #e20087; font-weight: =
500; margin: 0 0 20px; font-size: 14px; margin-top: 50px;\\">Data =
protection</h5>
        <small style=3D\\"color: #666; display: block; =
font-size: 11px; margin-bottom: 10px;\\">The Big Lottery Fund is a public =
body with a duty to distribute National Lottery and other money in grants =
for good causes. We use the personal data you provide, such as contact =
details for individuals at your organisation, to help you apply for a grant=
 and to assess your application.</small>
        <small style=3D\\"color: =
#666; display: block; font-size: 11px; margin-bottom: 10px;\\">We may share =
your personal data with organisations which help us with our grant making =
activities or others which have a legitimate interest in our work or have =
funded your grant. We will only share personal data which they need to =
carry out their work and subject to appropriate safety measures.</small>
        <small style=3D\\"color: #666; display: block; font-size: 11px; =
margin-bottom: 10px;\\">Our Data Protection and Privacy Notice gives more =
information about how we store and use personal data and the lawful basis =
for this. Please read the full Notice which is published on our website at =
www.biglotteryfund.org.uk/data-protection or contact us to request a hard =
copy. The Notice may be updated from time to time.</small>
    </body>
</html>

"
`;
